src/core/netif.cpp

```diff
@@ -147,17 +147,26 @@ struct netif *otrGetNetif(void)
     return &sNetif;
 }
 
-static void addAddress(const otIp6Address &aAddress)
+static void addAddress(const otIp6Address &aAddress, bool isMulticast)
 {
     otError error = OT_ERROR_NONE;
     err_t   err   = ERR_OK;
 
     LOCK_TCPIP_CORE();
+
     if (IsLinkLocal(aAddress))
     {
         netif_ip6_addr_set(&sNetif, 0, reinterpret_cast<const ip6_addr_t *>(&aAddress));
         netif_ip6_addr_set_state(&sNetif, 0, IP6_ADDR_PREFERRED);
     }
+    else if (isMulticast)
```

Comment:
nit: is there a reason to overload the existing `addAddress()` and `delAddress()` methods? why not have a separate `addUnicastAddress()` and `addMulticastAddress()` methods?